src/messaging/MessageCounterSync.cpp

```diff
@@ -241,14 +368,24 @@ void MessageCounterSyncMgr::HandleMsgCounterSyncResp(Messaging::ExchangeContext
     VerifyOrExit(resplen == kMsgCounterSyncRespMsgSize, err = CHIP_ERROR_INVALID_MESSAGE_LENGTH);
 
     syncCounter = chip::Encoding::LittleEndian::Read32(resp);
+    VerifyOrExit(syncCounter != 0, err = CHIP_ERROR_READ_FAILED);
+
     memcpy(challenge, resp, kMsgCounterChallengeSize);
 
     // Verify that the response field matches the expected Challenge field for the exchange.
     VerifyOrExit(memcmp(exchangeContext->GetChallenge(), challenge, kMsgCounterChallengeSize) == 0,
                  err = CHIP_ERROR_INVALID_SIGNATURE);
 
-    // ToDo:(#4628)Initialize/synchronize peer's message counter to FabricState.
-    VerifyOrExit(syncCounter != 0, err = CHIP_ERROR_READ_FAILED);
+    VerifyOrExit(packetHeader.GetSourceNodeId().HasValue(), err = CHIP_ERROR_INVALID_ARGUMENT);
+    peerNodeId = packetHeader.GetSourceNodeId().Value();
+
+    // Find an active connection to the specified peer node
+    state = mExchangeMgr->GetSessionMgr()->GetPeerConnectionState({ peerNodeId, packetHeader.GetEncryptionKeyID(), 0 });
```

Comment:
Creating a secure session state using KeyId is deprecated.

Please use `exchangeContext->GetSecureSessionHandle()` instead

Reply:
Acked